Src/ReadBarometer.c

```diff
@@ -6,127 +17,244 @@
         temperatureReading += dataInBuffer;
 
+        // Release chip select to end the command
         HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_SET);
 
-        // calculate 1st order pressure and temperature (1st order algorithm from MS5607-02BA03 datasheet)
-        double dT = temperatureReading - c5Tref * pow(2, 8);
-        double temp = 2000 + dT * c6Tempsens / pow(2, 23);
-        double off = c2Off * pow(2, 17) + dT * c4Tco / pow(2, 6);
-        double sens = c1Sens * pow(2, 16) + dT * c3Tcs / pow(2, 7);
-        // pressure and temperature were not divided by 100 to keep the decimal places
-        double pressure = (pressureReading * sens / pow(2, 21) - off) / pow(2, 15); // need to divide P by 100 to get mbar
-        double temperature = (temp); // need to divide T by 100 to get degrees Celcius
+        /* Calculate First-Order Temperature and Pressure --------------------*/
+
+        // Equations provided by MS5607-02BA03 data sheet
+
+        int32_t dT      = temperatureReading - (c5Tref << 8);
+        int32_t temp    = 2000 + ((dT * c6Tempsens) >> 23);                 // Divide this value by 100 to get degrees Celcius
+        int64_t off     = (c2Off << 17) + ((dT * c4Tco) >> 6);
+        int64_t sens    = (c1Sens << 16) + ((dT * c3Tcs) >> 7);
+        int32_t p       = (((pressureReading * sens) >> 21) - off) >> 15;   // Divide this value by 100 to get millibars
+
+        /* Calculate Second-Order Temperature and Pressure -------------------*/
+
+        // Equations provided by MS5607-02BA03 data sheet
+
+        if (temp < TEMP_LOW)    // If the temperature is below 20°C
+        {
+            int32_t t2      = (dT * dT) >> 31;
+            int32_t off2    = 61 * (((temp - 2000) * (temp - 2000)) >> 4);
+            int32_t sens2   = 2 * ((temp - 2000) * (temp - 2000));
+
+            if (temp < TEMP_VERY_LOW)   // If the temperature is below -15°C
+            {
+                off2    = off2 + (15 * ((temp + 1500) * (temp + 1500)));
+                sens2   = sens2 + (8 * ((temp + 1500) * (temp + 1500)));
+            }
+
+            temp    = temp  - t2;
+            off     = off   - off2;
+            sens    = sens  - sens2;
+        }
+
+        /* Store Data --------------------------------------------------------*/
 
-        if (osMutexWait(data->mutex_, 0) != osOK)
+        if (osMutexWait(data->mutex_, 0) == osOK)
         {
-            continue;
+            // Assign new values if mutex was available
+            data->pressure_     = p;
+            data->temperature_  = temp;
+            osMutexRelease(data->mutex_);
         }
 
-        data->pressure_ = (int32_t) pressure;
-        data->temperature_ = (int32_t) temperature;
-        osMutexRelease(data->mutex_);
+        // PRESSURE AND TEMPERATURE VALUES ARE STORED AT 100x VALUE TO MAINTAIN
+        // 2 DECIMAL POINTS OF PRECISION AS AN INTEGER!
+        // E.x. The value 1234 should be interpreted as 12.34
     }
 }
 
+/**
```

Comment:
I like the idea of using javadocs style comments for functions. If this comment describes what the function does, we don't need to repeat it in the code.
```
/**
 * This function reads and returns a 16-bit coefficient from the barometer.
 * @param   PROM_READ_CMD   The command to send in order to read the desired
 *                          coefficient from PROM. See data sheet for commands.
 * @return                  The read coefficient.
 */
uint16_t readCalibrationCoefficient(const uint8_t PROM_READ_CMD)
{
    uint16_t    coefficient;
    uint8_t     dataInBuffer;

    HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_RESET);

    HAL_SPI_Transmit(&hspi2, &PROM_READ_CMD, CMD_SIZE, CMD_TIMEOUT);

    // Read the first byte (bits 15-8)
    HAL_SPI_TransmitReceive(&hspi2, &READ_BYTE_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
    coefficient = dataInBuffer << 8;

    // Read the second byte (bits 7-0)
    HAL_SPI_TransmitReceive(&hspi2, &READ_BYTE_CMD, &dataInBuffer, CMD_SIZE, CMD_TIMEOUT);
    coefficient += dataInBuffer;

    HAL_GPIO_WritePin(BARO_CS_GPIO_Port, BARO_CS_Pin, GPIO_PIN_SET);

    return coefficient;
}
```

Reply:
Code changed to match